gymnasium/envs/toy_text/cliffwalking.py

```diff
@@ -18,39 +18,60 @@
 
 class CliffWalkingEnv(Env):
     """
-    This is a simple implementation of the Gridworld Cliff
-    reinforcement learning task.
+    Cliff walking involves crossing a gridworld from start to goal while avoiding falling off a cliff.
 
-    Adapted from Example 6.6 (page 106) from [Reinforcement Learning: An Introduction
-    by Sutton and Barto](http://incompleteideas.net/book/bookdraft2018jan1.pdf).
+    ## Description
+    The game starts with the player at location [3, 0] of the 4x12 grid world with the
+    goal located at [3, 11]. If the player reaches the goal the episode ends.
+
+    A cliff runs along [3, 1..10]. If the player moves to a cliff location it
+    returns to the start location.
+
+    The player makes moves until they reach the goal.
+
+    Adapted from Example 6.6 (page 132) from Reinforcement Learning: An Introduction
+    by Sutton and Barto [<a href="#ref">1</a>].
 
     With inspiration from:
     [https://github.com/dennybritz/reinforcement-learning/blob/master/lib/envs/cliff_walking.py](https://github.com/dennybritz/reinforcement-learning/blob/master/lib/envs/cliff_walking.py)
 
-    ## Description
-    The board is a 4x12 matrix, with (using NumPy matrix indexing):
-    - [3, 0] as the start at bottom-left
-    - [3, 11] as the goal at bottom-right
-    - [3, 1..10] as the cliff at bottom-center
-
-    If the agent steps on the cliff, it returns to the start.
-    An episode terminates when the agent reaches the goal.
-
-    ## Actions
-    There are 4 discrete deterministic actions:
-    - 0: move up
-    - 1: move right
-    - 2: move down
-    - 3: move left
-
-    ## Observations
-    There are 3x12 + 1 possible states. In fact, the agent cannot be at the cliff, nor at the goal
-    (as this results in the end of the episode).
-    It remains all the positions of the first 3 rows plus the bottom-left cell.
-    The observation is simply the current position encoded as [flattened index](https://numpy.org/doc/stable/reference/generated/numpy.unravel_index.html).
+    ## Action Space
+    The action shape is `(1,)` in the range `{0, 3}` indicating
+    which direction to move the player.
+
+    - 0: Move up
+    - 1: Move right
+    - 2: Move down
+    - 3: Move left
+
+    ## Observation Space
+    There are 3 x 12 + 1 possible states. The player cannot be at the cliff, nor at
```

Comment:
This is outside the scope of this pr, but should we be interested in adding an output with the coordinates of the agent. This could be an additional constructor argument and would change the output input, but would be easier to debug and as a demonstration to new users than an integer representing the state.

Reply:
I agree it would be easier to understand and remove a layer of complexity for new users. It took me a while to figure out how it works. FYI: taxi and frozen lake use the same approach.

Reply:
Ok, I don't think, this can be done in a future PR, possibly in the same PR as the transition probability PR